### PR TITLE
feat(config): Make RefreshInterval configurable

### LINKED ISSUE #509: Persist refresh interval on change
### Describe the feature request

In the ui part of the YAML config I want to set the default refresh value of the frontend. 

### Why do you personally want this feature to be implemented?

I am to lazy to switch to the 10s refresh interval again and again, a preconfigured one would be nice.

### How long have you been using this project?

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by macmoritz**
## Summary
This PR adds a config entry for the default value of the refresh interval of the ui.
Default is 300s/5m as before
Closes #509

## Checklist
- [x] Tested and/or added tests to validate that the changes work as intended, if applicable.
- [x] Updated documentation in `README.md`, if applicable.

### CODE REVIEW

File: config/ui/ui.go [hunk 1]
```diff
@@ -22,12 +23,13 @@ var (
...
-	Logo        string   `yaml:"logo,omitempty"`        // Logo to display on the page
-	Link        string   `yaml:"link,omitempty"`        // Link to open when clicking on the logo
-	Buttons     []Button `yaml:"buttons,omitempty"`     // Buttons to display below the header
+	Title           string   `yaml:"title,omitempty"`           // Title of the page
+	Description     string   `yaml:"description,omitempty"`     // Meta description of the page
+	Header          string   `yaml:"header,omitempty"`          // Header is the text at the top of the page
+	Logo            string   `yaml:"logo,omitempty"`            // Logo to display on the page
+	Link            string   `yaml:"link,omitempty"`            // Link to open when clicking on the logo
+	Buttons         []Button `yaml:"buttons,omitempty"`         // Buttons to display below the header
+	RefreshInterval int      `yaml:"refreshInterval,omitempty"` // RefreshInterval is the interval in seconds at which the page gets refreshed
```
Comment (TwiN) [inline config/ui/ui.go:32]:
```suggestion
	RefreshInterval int      `yaml:"refresh-interval,omitempty"` // Interval in seconds at which the page gets refreshed
```
For consistency

File: README.md [hunk 2]
```diff
@@ -240,6 +240,7 @@ If you want to test it locally, see [Docker](#docker).
 | `ui.header`                                     | Header at the top of the dashboard.                                                                                                         | `Health Status`            |
 | `ui.logo`                                       | URL to the logo to display.                                                                                                                 | `""`                       |
 | `ui.link`                                       | Link to open when the logo is clicked.                                                                                                      | `""`                       |
+| `ui.refreshInterval`                            | Default interval to refresh the page. Available values: 10, 30, 60, 120, 300, 600 (in seconds)                                              | `300`                      |
```
Comment (TwiN) [inline README.md:243]:
```suggestion
| `ui.refresh-interval`                            | Default interval to refresh the page. Available values: 10, 30, 60, 120, 300, 600 (in seconds)                                              | `300`                      |
```

In hindsight, perhaps this variable should be named `default-refresh-interval` instead 🤔

**Review by TwiN — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (macmoritz) [reply README.md:243]: Just changing the naming of the yaml key?
I would say so, otherwise we have a conflict in the go code with the default value.

Comment (macmoritz) [reply README.md]: Changed only the yaml key to `default-refresh-interval` cause of the conflicts mentioned above.

**Comment by macmoritz** [comment]
@TwiN I changed the naming of the YAML key. Could you rereview and give me some feedback or merge?

File: config/ui/ui.go [hunk 3]
```diff
@@ -22,12 +23,13 @@ var (
...
-	Logo        string   `yaml:"logo,omitempty"`        // Logo to display on the page
-	Link        string   `yaml:"link,omitempty"`        // Link to open when clicking on the logo
-	Buttons     []Button `yaml:"buttons,omitempty"`     // Buttons to display below the header
+	Title           string   `yaml:"title,omitempty"`                    // Title of the page
+	Description     string   `yaml:"description,omitempty"`              // Meta description of the page
+	Header          string   `yaml:"header,omitempty"`                   // Header is the text at the top of the page
+	Logo            string   `yaml:"logo,omitempty"`                     // Logo to display on the page
+	Link            string   `yaml:"link,omitempty"`                     // Link to open when clicking on the logo
+	Buttons         []Button `yaml:"buttons,omitempty"`                  // Buttons to display below the header
+	RefreshInterval int      `yaml:"default-refresh-interval,omitempty"` // Default Interval in seconds at which the page gets refreshed
```
Comment (TwiN) [inline config/ui/ui.go:32]: Can you rename `RefreshInterval` to `DefaultRefreshInterval` as well?

**Review by TwiN — CHANGES_REQUESTED** [review]

File: config/ui/ui.go (hunk 3 above)
Comment (macmoritz) [reply config/ui/ui.go:32]: I though about it, but there would be a naming conflict with the in-code default value:
```go
const (
    ...
    defaultRefreshInterval = 300
)
```
The only difference would be the _d_, this is confusing in my point of view.
Is there a better naming?

Comment (TwiN) [reply config/ui/ui.go:32]: variables/constants/types/methods/functions starting with a lowercase letter are not exported outside the package, so it's not too unusual imo.

The `default-refresh-interval` in the YAML is to make it more obvious for users that this value can be overriden, while the one from the constant is just for us. Honestly, you could remove the constant and set the value directly to 300 on L76 if you prefer.

Comment (macmoritz) [reply config/ui/ui.go]: Fixed

**Comment by macmoritz** [comment]
Hey @TwiN, could you please check again and update me?

**Comment by TwiN** [comment]
@macmoritz I know it's been a while since this has been open, but looking at the issue you created, why not just replace sessionStorage by localStorage instead?

https://github.com/TwiN/gatus/blob/bde30b2efb82982e6b77592cc0c690760906e8d1/web/app/src/components/Settings.vue#L36

https://github.com/TwiN/gatus/blob/bde30b2efb82982e6b77592cc0c690760906e8d1/web/app/src/components/Settings.vue#L81

I'm suggesting this, because it seems I accidentally used sessionStorage instead & that's why you have to keep re-setting the interval 😅 

Let me see if I can do something for you

**Comment by TwiN** [comment]
I believe I've implemented a satisfactory fix for #509 - let me know what you think.

I do appreciate you attempting to fix it via a PR, though!

**Comment by macmoritz** [comment]
This looks like a proper solution. Thanks for fixin.

**Comment by TwiN** [comment]
@macmoritz Glad you're satisfied! Thanks again for contributing.

**Pull request closed without merging**